src/instant-test/instant-test.component.js

```diff
@@ -1,31 +1,33 @@
-import React from "react";
+import React, { useState } from "react";
 import { LocalStorageContext } from "../shared/use-local-storage-data.hook";
 
 export default function InstantTest(props) {
-  const params = new URLSearchParams(location.search);
-  const isValid = params.has("name") && params.has("url");
+  const [isValid, setIsValid] = useState(true);
   const { addApplication } = React.useContext(LocalStorageContext);
 
   React.useEffect(() => {
-    if (!isValid) {
-      return;
-    }
+    const params = new URLSearchParams(location.search);
 
-    const app = {
-      name: params.get("name"),
-      pathPrefix: "/",
-      framework: params.get("framework") || undefined,
-    };
+    if (!(params.has("name") && params.has("url"))) {
+      setIsValid(false);
+    } else {
+      const app = {
+        name: params.get("name"),
+        pathPrefix: "/",
+        framework: params.get("framework") || undefined,
+      };
 
-    addApplication(app);
+      addApplication(app);
 
-    window.importMapOverrides.addOverride(
-      params.get("name"),
-      params.get("url")
-    );
+      window.importMapOverrides.addOverride(
+        params.get("name"),
+        params.get("url")
+      );
 
-    location.assign("/");
-  }, [addApplication, isValid, params]);
+      location.assign("/");
```

Comment:
@joeldenning I believe this line here makes a full page reload when adding Angular for the first time so that ZoneJS gets included (in reference to https://github.com/single-spa/create-single-spa/pull/135#discussion_r433392042). Tested this locally with a generated Angular application and it works well.

Reply:
Yes, this will cause a page reload. Good catch.